src/Arcus.Security.Core/Exceptions/SecretNotFoundException.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Arcus.Security.Core.Exceptions
+{
+    /// <summary>
+    /// Exception, thrown when no secret was found, using the given name.
+    /// </summary>
+    public class SecretNotFoundException : Exception
```

Comment:
Good to have custom Exceptions but they should be marked as `[Serializable]` to conform with the guidelines

Reply:
Agree !  Fixed with new commit.